### ISSUE TITLE
Improve error messages for OCD-IDs

### ISSUE THREAD

**Issue by jdmgoogle** [google/election_results_xml_validator#34]
Improve the error messages surrounding OCD-IDs to distinguish between the following cases:

1. GpUnit doesn't have any external identifier
2. GpUnit has an ocd-id but it does not have a valid value
3. GpUnit has an external identifier but the the type is not set as "ocd-id" (could be "OCD-ID", in which case saying that that GpUnit doesn't have a valid ocd-id could be a bit confusing).

**Comment by miano** [comment]
Addressed 1 and 2 in the commit above. 3 should be caught by the schema rule since the restriction in the schema is case sensitive(see below). Relaxed it to the rule to be case-insensitive nonetheless.

```
<xs:simpleType name="IdentifierType">
        <xs:restriction base="xs:string">
            <xs:enumeration value="fips" />
            <xs:enumeration value="local-level" />
            <xs:enumeration value="national-level" />
            <xs:enumeration value="ocd-id" />
            <xs:enumeration value="state-level" />
            <xs:enumeration value="other" />
        </xs:restriction>
</xs:simpleType>
```
